Dropdown/Dropdown.js

```diff
@@ -0,0 +1,200 @@
+ * </Dropdown>
+ *
+ * @module agate/Dropdown
+ * @exports Dropdown
+ */
+
+import React from 'react';
+import kind from '@enact/core/kind';
+import Changeable from '@enact/ui/Changeable';
+import Group from '@enact/ui/Group';
+import Icon from '@enact/agate/Icon';
+import Item from '@enact/agate/Item';
+import Toggleable from '@enact/ui/Toggleable';
+import Transition from '@enact/ui/Transition';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import Skinnable from '../Skinnable';
+import SpotlightContainerDecorator from '@enact/spotlight/SpotlightContainerDecorator';
+import Scroller from '@enact/ui/Scroller';
+import componentCss from './Dropdown.module.less';
+
+const ContainerDiv = SpotlightContainerDecorator({enterTo: 'last-focused'}, 'div');
+const isSelectedValid = ({children, selected}) => Array.isArray(children) && children[selected] != null;
+
+/**
+ * A stateless Dropdown component.
+ *
+ * @class DropdownBase
+ * @memberof agate/Dropdown
+ * @ui
+ * @public
+ */
+const DropdownBase = kind({
+	name: 'Dropdown',
+
+	propTypes: /** @lends agate/Dropdown.Dropdown.prototype */ {
+		/*
+		 * The selections for Dropdown
+		 *
+		 * @type {String[]|Array.<{key: (Number|String), children: (String|Component)}>}
+		 */
+		children: PropTypes.array,
+
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		css: PropTypes.object,
+
+		/**
+		 * The direction where the dropdown list appears.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		direction: PropTypes.string,
+
+		/*
+		 * Called when an item is selected.
+		 *
+		 * @type {Function}
+		 */
+		onSelect: PropTypes.func,
+
+		/*
+		 * Called when clicked on the dropdown to open.
+		 *
+		 * @type {Number}
+		 *
+		 */
+		open: PropTypes.bool,
+
+		/*
+		 * Index of the selected item.
+		 *
+		 * @type {Number}
+		 *
+		 */
+		selected: PropTypes.number
+	},
+
+	defaultProps: {
+		direction: 'down',
+		open: false
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'dropdown',
+		publicClassNames: true
+	},
+
+	computed: {
+		transitionContainerClassname: ({css, open, direction, styler}) => styler.join(css.transitionContainer, {openTransitionContainer: open, upTransitionContainer: direction === 'up'} ),
+		dropdownListClassname: ({children, css, styler}) => styler.join(css.dropdownList, {dropdownListWithScroller: children.length > 4}),
+		title: ({children, selected}) => {
+			if (isSelectedValid({children, selected})) {
+				const child = children[selected];
+				return typeof child === 'object' ? child.children : child;
+			}
+
+			return children[0];
+		},
+		transitionDirection: ({direction}) => {
+			switch (direction) {
+				case 'left':
+					return 'right';
+				case 'right':
+					return 'left';
+				case 'up':
+					return 'down';
+				case 'down':
+					return 'up';
```

Comment:
Should we add `default:` here?

Reply:
I think this is the opposite of what we would want, isn't it?  The transition is opposite from the opening so it should be the same as the default.